packages/backend-auth/src/index.ts

```diff
@@ -1,4 +1,6 @@
 export { BackendAuth, AmplifyAuthProps, defineAuth } from './factory.js';
+// ReferenceAuthConstruct is intentionally not public, but props are required
+export { ReferenceAuthProps } from './reference_construct.js';
```

Comment:
Instead of this comment.
I'd move the type to `reference_factory.ts`. 
To underscore that it's primary purpose is to type the input to `referenceAuth`.
It can then be then used in construct as well.
If this happen to created import cycle then right place might be `types.ts`

Reply:
Updated, moved to reference_factory to make it more clear